mmv1/third_party/terraform/services/bigquery/data_source_google_bigquery_tables.go

```diff
@@ -0,0 +1,143 @@
+package bigquery
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleBigQueryTables() *schema.Resource {
+
+	dsSchema := map[string]*schema.Schema{
+		"dataset_id": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The ID of the dataset containing the tables.",
+		},
+		"project": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The ID of the project in which the dataset is located. If it is not provided, the provider project is used.",
+		},
+		"tables": {
+			Type:     schema.TypeList,
+			Computed: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"labels": {
```

Comment:
We don't need to add this for now since there is no use case of it yet, I just mentioned it in a previous comment as an example. Let's remove it to scope down the PR but thanks for looking into it.

Reply:
Hey @wj-chen! Appreciate the feedback here.

I was hoping that we keep this particular piece in (if acceptable/reasonable to of course). Having this available OOB will be of immense value to my organization, as there's a lot of power in leveraging the labels applied to tables for conditionally performing operations.

Would this be possible to leave in by any chance?